enterprise/frontend/src/metabase-enterprise/audit_app/components/QueryViewer/QueryViewer.tsx

```diff
@@ -45,7 +42,8 @@ export function QueryViewer({ datasetQuery }: { datasetQuery: DatasetQuery }) {
     );
   }
 
-  if (query.tables()) {
+  const database = question.database();
+  if (database && database.tables) {
```

Comment:
This is exactly the same check as in `query.tables()` in `StructuredQuery`. Not 100% check what this check does but I want to leave it as it is.